compiler/rustc_expand/src/expand.rs

```diff
@@ -1060,13 +1060,26 @@ impl<'a, 'b> InvocationCollector<'a, 'b> {
     // since they will not be detected after macro expansion.
     fn check_attributes(&mut self, attrs: &[ast::Attribute]) {
         let features = self.cx.ecfg.features.unwrap();
-        for attr in attrs.iter() {
+        let mut attrs = attrs.iter().peekable();
+        let mut span: Option<Span> = None;
+        while let Some(attr) = attrs.next() {
             rustc_ast_passes::feature_gate::check_attribute(attr, self.cx.sess, features);
             validate_attr::check_meta(&self.cx.sess.parse_sess, attr);
+
+            if let Some(sp) = span {
+                span = Some(sp.to(attr.span));
+            } else {
+                span = Some(attr.span);
+            }
```

Comment:
You can remove the `unwrap()` below by adding a local variable:
```suggestion
            let current_span = if let Some(sp) = span {
                sp.to(attr.span)
            } else {
                attr.span
            };
            span = Some(current_span);
```